pgrx-examples/composite_type/src/lib.rs

```diff
@@ -145,6 +145,7 @@ fn make_friendship(
 }
 
 /*
+FIXME: make this example no longer DOA
```

Comment:
What's the plan for this?  Aggregate support is a pretty important feature

Reply:
The problem isn't with Aggregate, per se, it's with certain types.

These are aggregates that are supposed to manage composite types. But we know composite types are allocated in Postgres, thus are effectively lifetime-bound. But Aggregate doesn't know how to convey the lifetimes through. Thus even annotating it with something like this does not actually fix the problem:
```rust
#[pg_aggregate]
impl<'a> Aggregate for SumScritches<'a> {
    type State = i32;
    const INITIAL_CONDITION: Option<&'static str> = Some("0");
    type Args = pgrx::name!(value, pgrx::composite_type!('a, "Dog"));

    fn state(
        current: Self::State,
        arg: Self::Args,
        _fcinfo: pg_sys::FunctionCallInfo,
    ) -> Self::State {
        todo!()
    }
}
```

Thus in order to make aggregates work for complex cases we must break the world for them as well, making all their functions use the correct, lifetime-bound types. Hopefully this will let us simplify the code expansion for them as well.